src/modules/currencies/currencies.controller.ts

```diff
@@ -0,0 +1,77 @@
+import { Controller, Get, Param, Query } from '@nestjs/common';
+import { ApiBearerAuth, ApiOperation, ApiParam, ApiQuery, ApiResponse, ApiTags } from '@nestjs/swagger';
+
+import { CurrenciesService } from './currencies.service';
+import { CurrencyDto, GetCurrencyExchangeDto } from './dto';
+import { CurrencyEntity } from './entities/currency.entity';
+import { CurrencyExchangeEntity } from './entities/currency-exchange.entity';
+
+@ApiBearerAuth()
+@ApiTags('currencies')
+@Controller('currencies')
+export class CurrenciesController {
+  constructor(private readonly currenciesService: CurrenciesService) {}
+
+  @Get()
+  @ApiResponse({
+    status: 200,
+    description: 'Get all currencies',
+  })
+  findAll(): Promise<CurrencyEntity[]> {
+    return this.currenciesService.findAll();
+  }
+
+  @Get('exchange')
+  @ApiOperation({
+    summary: 'Get the Active exchange rate based on source and target exchange rates. Query parameters are mandatory for this endpoint.',
+  })
+  @ApiResponse({
+    status: 200,
+    description: 'Get the Active exchange rate',
+    type: GetCurrencyExchangeDto,
+  })
+  @ApiQuery({
+    name: 'source',
+    required: true,
+    type: 'string',
+    description: 'Source currency for exchange rate - Use ISO 3 alpha currency code standard',
```

Comment:
We support just 2 source currencies, GBP and USD. Maybe this should be part of swagger documentation and/or validation.

Reply:
good shout - updated now